Added instructions for running composer using docker (#6481)
magento/devdocs: src/cloud/docker/docker-config.md

{: .bs-callout-warning}
This command removes all components of your local Docker instance including containers, networks, volumes, and images except for the persistent database and the `magento-sync` volume. See [Rebuild a clean environment][refresh].

## Running Composer with Docker

> It is possible to run composer using `docker` before you have instanced the containers. This is very useful for instancing the application during the CI/CD build process, or even first time Magento set up.

Reviewer: ```suggestion
You can run composer using `docker` commands before you create the containers. This technique is useful for creating an application instance during the CI/CD build process, or even first time Magento set up.
```

> To do this you must use the proper [Docker Hub PHP Image Tag] that matches with the version of the Magento application. The following example shows PHP 7.3 and should be run in the project root directory.

Reviewer: ```suggestion
When you run composer with Docker commands, you must use the [Docker Hub PHP Image Tag] that matches the Magento application version. The following example uses PHP 7.3. Run the command from the project root directory. 
```

```bash
docker run -it  -v $(pwd):/app/ -v ~/.composer/:/root/.composer/ magento/magento-cloud-docker-php:7.3-cli-1.1 bash -c "composer install&&chown www. /app/"
```

> This command passes in the current working directory as `/app/`, includes composer from `~/.composer/` and runs the `composer install` in the container, after which it fixes the permissions on the files that have been added/changed.

Reviewer: ```suggestion
This command passes in the current working directory as `/app/`, includes composer from `~/.composer/`, and runs the `composer install` command in the container. After this set up, the command  fixes the permissions on the files that have been added or changed.
```
